spec/models/accounting/quickbooks/updater_spec.rb

```diff
@@ -41,19 +57,121 @@
       'total' => '19.99',
       'private_note' => 'Nate now testing' }
   end
+  let(:txn) { create(:accounting_transaction, project: loan, quickbooks_data: quickbooks_data) }
+  let!(:line_items) do
+    txn.line_items = [create(:line_item,
+        qb_line_id: 0,
+        description: 'Nate desc',
+        amount: '15.09',
+        account: create(:account, name: 'Accounts Receivable (A/R)'),
+        posting_type: 'Debit'),
+      create(:line_item,
+        qb_line_id: 1,
+        description: 'Nath desc',
+        amount: '12.37',
+        account: prin_acct,
+        posting_type: 'Credit'),
+      create(:line_item,
+        qb_line_id: 2,
+        description: 'Jay desc',
+        amount: '2.72',
+        account: int_rcv_acct,
+        posting_type: 'Credit')]
+  end
 
+  subject { described_class.new(connection) }
 
   before do
     allow(subject).to receive(:service).and_return(generic_service)
     allow(connection).to receive(:update_attribute).with(:last_updated_at, anything)
   end
 
-  subject { described_class.new(connection) }
+  context 'QB line item manipulations' do
+    context 'line item added' do
+
+      before do
+        quickbooks_data['line_items'] << {
+          'id' => '3',
+          'description' => 'Jazzy desc',
+          'amount' => '1.0',
+          'detail_type' => 'JournalEntryLineDetail',
+          'journal_entry_line_detail' => {
+            'posting_type' => 'Debit',
+            'entity' => {
+              'type' => 'Customer',
+              'entity_ref' => { 'value' => '1', 'name' => "Amy's Bird Sanctuary", 'type' => nil } },
+            'account_ref' => { 'value' => '84', 'name' => int_rcv_acct.name, 'type' => nil },
+            'class_ref' => { 'value' => '5000000000000026437', 'name' => loan.id, 'type' => nil },
+            'department_ref' => nil } }
+
+        txn.update(quickbooks_data: quickbooks_data)
+        subject.send(:extract_qb_data, txn)
+      end
+
+      it 'updates correctly in Madeline' do
+        expect(txn.reload.line_items.count).to eq(4)
+        expect(txn.reload.line_items.last.qb_line_id).to eq(3)
```

Comment:
You shouldn't need to reload more than once unless you changed something in between.